examples/platform_channel/macos/Runner/PowerSource.swift

```diff
@@ -0,0 +1,78 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import IOKit.ps
+
+enum PowerState {
+  case ac
+  case battery
+  case unknown
+}
+
+/// A convenience wrapper for an IOKit power source.
+class PowerSource {
+  let info = IOPSCopyPowerSourcesInfo().takeRetainedValue()
+  lazy var sources = IOPSCopyPowerSourcesList(info).takeRetainedValue() as Array
+
+  /// Returns the current power source capacity. Apple-defined power sources will return this value
+  /// as a percentage.
+  func getCurrentCapacity() -> Int {
```

Comment:
can return `Int?` rather than `-1` to indicate NA capacity. This way the caller won't forget to deal with -1 case

Reply:
The current interface is intentional for consistency with iOS, Android, Windows, and Linux implementations, and to keep the diffs in the method handler code as minimal as possible.

That said - I think combined with the `guard` statement it's a really nice improvement and worth the difference. Long-term the goal is to make Swift default for iOS and Kotlin for Android, both of which have non-primitive/nullable `Int`s, so I'd expect we can move all the examples over and someone's got to be first.

The current iOS Swift sample is unnecessarily different from the others in that it conflates fetching battery level and reporting battery level, but I'll tweak that in the same patch.